ui/src/app/pages/workspace/workspace-card.tsx

```diff
@@ -171,32 +175,13 @@ export const WorkspaceCard = fp.flow(withNavigation)(
         : triggerEvent(EVENT_CATEGORY, 'navigate', 'Click on workspace name');
     }
 
-    onClick() {
-      const {
-        workspace: { id, namespace },
-      } = this.props;
+    onClick(e) {
       if (this.requiresReviewPrompt()) {
         this.setState({ showResearchPurposeReviewModal: true });
-      } else {
-        this.trackWorkspaceNavigation();
-        this.props.navigate(['workspaces', namespace, id, 'data']);
+        e.preventDefault();
       }
     }
 
-    getRightClickTarget() {
-      // Restrict open new tab option/right click options, if workspace requires review
-      if (this.requiresReviewPrompt()) {
-        return;
-      }
-
-      const {
-        workspace: { id, namespace },
-      } = this.props;
-
-      this.trackWorkspaceNavigation();
```

Comment:
This was the second regression. Note that this function is called whenever the workspace card link is rendered, which is at least once everytime the workspace card appears on the page. This means we were sending a ton of events where we shouldn't have.